src/transformers/models/vqgan/modeling_vqgan.py

```diff
@@ -0,0 +1,763 @@
+        codebook_loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `return_loss` is `True`):
+            The codebook loss to optimize.
+    """
+
+    quantized_states: torch.FloatTensor = None
+    codebook_indices: torch.LongTensor = None
+    codebook_loss: Optional[torch.FloatTensor] = None
+
+
+@dataclass
+class VQGANModelOutput(ModelOutput):
+    """
+    Args:
+        reconstructed_pixel_values (`torch.FloatTensor` of shape `(batch_size, channels, height, width)`):
+            The reconstructed pixel values.
+        codebook_indices (`torch.LongTensor` of shape `(batch_size, latent_height*latent_width)`):
+            The indices in the codebook (embedding) matrix for the `quantized_states`.
+        quantized_states (`torch.FloatTensor` of shape `(batch_size, channels, latent_height, latent_width)`):
+            The quantized states obtained by applying the `VectorQuantizer` to the output of `Encoder`.
+        codebook_loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `return_loss` is `True`):
+            The codebook loss to optimize.
+    """
+
+    reconstructed_pixel_values: torch.FloatTensor = None
+    quantized_states: torch.FloatTensor = None
+    codebook_indices: torch.LongTensor = None
+    codebook_loss: Optional[torch.FloatTensor] = None
+
+
+class Upsample(nn.Module):
+    def __init__(self, in_channels: int, with_conv: bool):
+        super().__init__()
+
+        self.with_conv = with_conv
+
+        if self.with_conv:
+            self.conv = nn.Conv2d(
+                in_channels,
+                in_channels,
+                kernel_size=3,
+                stride=1,
+                padding=1,
+            )
+
+    def forward(self, hidden_states):
+        hidden_states = torch.nn.functional.interpolate(hidden_states, scale_factor=2.0, mode="nearest")
+        if self.with_conv:
+            hidden_states = self.conv(hidden_states)
+        return hidden_states
+
+
+class Downsample(nn.Module):
+    def __init__(self, in_channels: int, with_conv: bool):
+        super().__init__()
+
+        self.with_conv = with_conv
+        if self.with_conv:
+            self.conv = nn.Conv2d(
+                in_channels,
+                in_channels,
+                kernel_size=3,
+                stride=2,
+                padding=0,
+            )
+
+    def forward(self, hidden_states):
+        if self.with_conv:
+            pad = (0, 1, 0, 1)  # pad height and width dim
+            hidden_states = torch.nn.functional.pad(hidden_states, pad, mode="constant", value=0)
+            hidden_states = self.conv(hidden_states)
+        else:
+            hidden_states = torch.nn.functional.avg_pool2d(hidden_states, kernel_size=2, stride=2)
+        return hidden_states
+
+
+class ResnetBlock(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int = None,
+        use_conv_shortcut: bool = False,
```

Comment:
this param is never used and always defaults to `False` -> let's remove it and also remove the corresponding `use_conv_short_cut` param